internal/nix/install.go

```diff
@@ -88,3 +89,43 @@ func DirExists() bool {
 func isRoot() bool {
 	return os.Geteuid() == 0
 }
+
+func EnsureNixInstalled(writer io.Writer, daemon *bool) error {
```

Comment:
a straight move, with the only change to line 120 `Install(writer, daemon)`. In the original code this was `cmd.ErrOrStderr, daemonFlag(..)` which is passed in from the caller.